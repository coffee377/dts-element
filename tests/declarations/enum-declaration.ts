import * as dts from '../index';

it('should return correctly with name', () => {
  expect(dts.emit(
    dts.create_enum_declaration({
      name: 'E',
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, export', () => {
  expect(dts.emit(
    dts.create_enum_declaration({
      name: 'E',
      export: true,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, members', () => {
  expect(dts.emit(
    dts.create_enum_declaration({
      name: 'E',
      members: [
        dts.create_variable_declaration({
          name: 'a',
          type: dts.create_literal_type({
            value: 5,
          }),
        }),
        dts.create_variable_declaration({
          name: 'b',
        }),
        dts.create_variable_declaration({
          name: 'c',
          type: dts.create_literal_type({
            value: 'hello',
          }),
        }),
      ],
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, export, members', () => {
  expect(dts.emit(
    dts.create_enum_declaration({
      name: 'E',
      export: true,
      members: [
        dts.create_variable_declaration({
          name: 'a',
          type: dts.create_literal_type({
            value: 5,
          }),
        }),
        dts.create_variable_declaration({
          name: 'b',
        }),
      ],
    }),
  )).toMatchSnapshot();
});

it('should throw error with members (VariableDeclaration with type (not undefined or LiteralType))', () => {
  expect(() => dts.emit(
    dts.create_enum_declaration({
      name: 'E',
      members: [
        dts.create_variable_declaration({
          name: 'a',
          type: dts.any_type,
        }),
      ],
    }),
  )).toThrowError();
});
