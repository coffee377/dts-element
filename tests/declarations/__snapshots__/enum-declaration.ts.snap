// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly with name 1`] = `
"enum E {
}"
`;

exports[`should return correctly with name, export 1`] = `
"export enum E {
}"
`;

exports[`should return correctly with name, export, members 1`] = `
"export enum E {
    a = 5,
    b = 6
}"
`;

exports[`should return correctly with name, members 1`] = `
"enum E {
    a = 5,
    b = 6,
    c = \\"hello\\"
}"
`;
